walk/config.go

```diff
@@ -166,12 +167,75 @@ func loadBazelIgnore(repoRoot string) (isIgnoredFunc, error) {
 		excludes[ignore] = struct{}{}
 	}
 
-	isIgnored := func(p string) bool {
+	isBazelIgnored := func(p string) bool {
 		_, ok := excludes[p]
 		return ok
 	}
 
-	return isIgnored, nil
+	return isBazelIgnored, nil
+}
+
+func loadRepoDirectoryIgnore(repoRoot string) (isIgnoredFunc, error) {
+	ignorePath := path.Join(repoRoot, "REPO.bazel")
+	repoFileContent, err := os.ReadFile(ignorePath)
+	if errors.Is(err, fs.ErrNotExist) {
+		return nothingIgnored, nil
+	}
+	if err != nil {
+		return nothingIgnored, fmt.Errorf("REPO.bazel exists but couldn't be read: %v", err)
+	}
+
+	ast, err := bzl.Parse(repoRoot, repoFileContent)
+	if err != nil {
+		return nothingIgnored, fmt.Errorf("failed to parse REPO.bazel: %v", err)
+	}
+
+	var ignoreDirectories []string
+
+	// Search for ignore_directories([...ignore strings...])
+	for _, expr := range ast.Stmt {
+		if call, isCall := expr.(*bzl.CallExpr); isCall {
+			if inv, isIdentCall := call.X.(*bzl.Ident); isIdentCall && inv.Name == "ignore_directories" {
+				if len(call.List) != 1 {
+					return nothingIgnored, fmt.Errorf("REPO.bazel ignore_directories() expects one argument")
+				}
+
+				list, isList := call.List[0].(*bzl.ListExpr)
```

Comment:
Users may be passing in a list comprehension. In that case, we probably shouldn't fail hard, but emit a warning.

Reply:
If it's not a list it just returns `nothingIgnored` and the returned error gets logged. That's the same as bazelignore.